webapp/src/components/Conference/Conference.tsx

```diff
@@ -28,26 +30,43 @@ export const Conference = (): JSX.Element => {
     setPipHidden(!pipHidden)
   }
 
+  const videoTracks = localStream?.getVideoTracks()
+  const videoTrackId = videoTracks != null && videoTracks.length !== 0 ? videoTracks[0].id : ''
+
+  // Only re-render the selfie if the videoTrack id changes
+  const selfie = useMemo((): JSX.Element => {
+    return <Selfview localMediaStream={localStream} isVideoInputMuted={false} shouldShowUserAvatar={false} username={''} data-testid='SelfView'/>
+  }, [videoTrackId])
+
+  const handleUserSettings = (userSettings: UserSettings): void => {
+    changeDevices(userSettings).catch((e) => { console.error(e) })
+  }
+
+  useEffect(() => {
+    // Subscribe to changes after each rendering
+    setUserSettingsEventListener(handleUserSettings)
+  })
+
```

Comment:
I dont quite get why we handle user settings state differently to other states. Should we decide if the state needs to be local or global and then we use the same pattern we use elsewhere. I see we have some global contexts we have some redux as well and we have this module variables. Should we agree on one way of doing things?

Reply:
Redux is only used for the Mattermost library. I tried to use the React Context, but the integration is tricky at some points. I refactored this code. Could you take a look to it?

Reply:
I still dont get why we need to have a `useEffect` which run some code each rerender, should all of this logic related to changing devices be outside of react and you just subscribe to changes or call a methods from external service if this code is global anyways?

Reply:
It was an easy way to pass a function that will be called each time the settings change in Mattermost.

Instead of defined a callback function, I can define an `EventEmitter` and subscribe/unsubscribe when mounting. What do you think?